src/apps/service/apps/jobs/sharing.clj

```diff
@@ -150,42 +161,94 @@
 
      (let [child-input-share-errs (process-child-jobs (partial share-child-job apps-client sharer sharee level) job-id)
            input-share-errs       (process-job-inputs (partial share-input-file sharer sharee) apps-client job)
            output-share-err-msg   (share-output-folder sharer sharee job)
            app-share-err-msg      (share-app-for-job apps-client sharer sharee job-id job)]
+       (update-fn (format "shared job ID %s with %s" job-id sharee))
        (job-sharing-success job-id
                             job
                             level
                             (concat input-share-errs child-input-share-errs)
                             output-share-err-msg
                             app-share-err-msg))
      (catch [:type ::job-sharing-failure] {:keys [failure-reason]}
+       (update-fn (format "failed to share job ID %s with %s: %s" job-id sharee failure-reason))
        (job-sharing-failure job-id job level failure-reason))
-     (catch [:type ::permission-load-failure] {:keys [reason]}
-       (job-sharing-failure job-id job level (job-sharing-msg :load-failure job-id reason))))
-    (job-sharing-failure job-id nil level (job-sharing-msg :not-found job-id))))
+     (catch Object _
+       (update-fn (format "failed to share job ID %s with %s: %s" job-id sharee (str (:throwable &throw-context))))
+       (job-sharing-failure job-id job level)))))
 
 (defn- share-jobs-with-user
-  [apps-client sharer {sharee :subject :keys [analyses]}]
-  (let [responses (mapv (partial share-job apps-client sharer sharee) analyses)]
+  [update-fn apps-client sharer {sharee :subject :keys [analyses]}]
+  (let [responses (mapv (partial share-job update-fn apps-client sharer sharee) analyses)]
     (cn/send-analysis-sharing-notifications (:shortUsername sharer) sharee responses)
     {:subject  sharee
      :analyses responses}))
 
+(defn- share-jobs-thread
+  [async-task-id]
+  (let [{:keys [username data]} (async-tasks/get-by-id async-task-id)]
+    (try+
+     (async-tasks/add-status async-task-id {:status "running"})
+     (let [apps-client (get-apps-client-for-username username)
+           user        (.getUser apps-client)]
+       (mapv (partial share-jobs-with-user (async-tasks/update-fn async-task-id "running") apps-client user)
+             (:sharing data))
+       (async-tasks/add-completed-status async-task-id {:status "completed"}))
+     (catch Object _
+       (log/error (:throwable &throw-context) "failed processing async task" async-task-id)
+       (async-tasks/add-completed-status async-task-id {:status "failed"})
+       (cn/send-general-analysis-sharing-failure-notification username async-task-id)))))
+
 (defn share-jobs
-  [apps-client user sharing-requests]
-  (mapv (partial share-jobs-with-user apps-client user) sharing-requests))
+  [apps-client {username :shortUsername} sharing-requests]
+  (-> (log/spy :warn (async-tasks/new-task "analysis-sharing" username sharing-requests))
```

Comment:
Debug logging that can be removed now?
```suggestion
  (-> (async-tasks/new-task "analysis-sharing" username sharing-requests)
```

Reply:
Ah, yep, I missed that. Thanks for catching it!